theseus/geometry/se3.py

```diff
@@ -0,0 +1,167 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Optional, Union, cast
+
+import torch
+
+import theseus
+import theseus.constants
+
+from .lie_group import LieGroup
+from .point_types import Point3
+from .so3 import SO3
+
+
+class SE3(LieGroup):
+    def __init__(
+        self,
+        x_y_z_quaternion: Optional[torch.Tensor] = None,
+        data: Optional[torch.Tensor] = None,
+        name: Optional[str] = None,
+        dtype: Optional[torch.dtype] = None,
+    ):
+        if x_y_z_quaternion is not None and data is not None:
+            raise ValueError("Please provide only one of x_y_z_quaternion or data.")
```

Comment:
Should we be more explicit here and for other Lie groups what we mean by data? For instance here it is a `bx4x4` transformation matrix.

Reply:
This line is only used to resolve the conflict where both `x_y_z_quaternion` and `data` are provided.

There is another [function](https://github.com/facebookresearch/theseus/blob/f8cf872dc3d3acffc3232eace5826c6ef486babb/theseus/geometry/se3.py#L62) that explicitly defines what `data` should look like.

Reply:
That makes sense. I meant, when this error statement is printed the user will have an opaque prompt for `data`.

Reply:
I think this is fine as is. The format for data will be documented and there are also other exceptions thrown if the user provides the wrong format for data.